File: solutions/ruby/lasagna/1/lasagna.rb
class Lasagna

  EXPECTED_MINUTES_IN_OVEN = 40
  MINUTES_FOR_PREPARATION = 2

  def remaining_minutes_in_oven(actual_minutes)
    EXPECTED_MINUTES_IN_OVEN - actual_minutes
  end

  def preparation_time_in_minutes(layers)
    MINUTES_FOR_PREPARATION * layers
  end

  def total_time_in_minutes(number_of_layers:, actual_minutes_in_oven:)
    preparation_time_in_minutes(number_of_layers) + actual_minutes_in_oven
  end

end

File: solutions/elixir/lasagna/2/lib/lasagna.ex
defmodule Lasagna do

  def expected_minutes_in_oven, do: 40

  def remaining_minutes_in_oven(actual_minutes),
      do: expected_minutes_in_oven() - actual_minutes

  def preparation_time_in_minutes(layers), do: 2 * layers

  def total_time_in_minutes(layers, actual_minutes),
    do: preparation_time_in_minutes(layers) + actual_minutes

  def alarm, do: "Ding!"

end
